server.go

```diff
@@ -286,6 +294,28 @@ func (s *Server) handleProxyError(err error, w http.ResponseWriter) {
 	w.Write([]byte("Internal Server Error"))
 }
 
+func (s *Server) GetRoute(ctx context.Context) *Route {
```

Comment:
```suggestion
func RouteFromContext(ctx context.Context) *Route {
```

Thoughts on this naming? (I somewhat copied this from the go docs examples)

Also, I think we can expose it as a function instead of calling it from the `Server` struct.

Reply:
Yeah, that naming sounds good to me, will update 👍